server/proxy/httpproxy/director.go

```diff
@@ -41,9 +42,13 @@ func newDirector(lg *zap.Logger, urlsFunc GetProxyURLs, failureWait time.Duratio
 		lg:          lg,
 		uf:          urlsFunc,
 		failureWait: failureWait,
+		stopc:       make(chan struct{}),
+		donec:       make(chan struct{}),
 	}
+	osutil.RegisterInterruptHandler(d.stop)
```

Comment:
Please remove this for now, and let upper layer to handle the interrupt handler. 

We might also be totally removing the v2 http proxy.